### PR TITLE
Various Quality of Life Improvements

### PR DESCRIPTION
**Pull request by Vinyzu**
- Added Order Generator
- Improved Code Flow
- Added New Features

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -101,6 +101,12 @@ ignore = [
     "S101",
     # Function Arguments
     "PLR0913",
+    # UTC Timezones
+    "DTZ003",
```
Comment (koviubi56) [inline pyproject.toml]: Working with timezones is a mess, so I think it's better to _always_ use timezone-aware UTC datetimes, that way we can't mess up.

File: src/sincere_singularities/modules/order_generator.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+        )
+        # Generating Foods
+        order = self._generate_menu(order, restaurant_name)
+        # Generating Order Description
+        order_description = self._generate_order_description(order, has_delivery_time, has_extra_information)
+
+        return order, order_description
+
+    def _generate_menu(self, order: Order, restaurant_name: str) -> Order:
+        restaurant = next(_restaurant for _restaurant in RESTAURANT_JSON if _restaurant.name == restaurant_name)
```
Comment (koviubi56) [inline src/sincere_singularities/modules/order_generator.py]: Use `points`' `get_restaurant_by_name`

File: src/sincere_singularities/modules/order_generator.py [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+        # Get Probability to Generate Multiple Dishes in one Order
+        multiple_dish_probabilities = MULTIPLE_DISH_PROBABILITY[self.difficulty]
+
+        # TODO: Don't add a Menu Item from every section everytime
```
Comment (koviubi56) [inline src/sincere_singularities/modules/order_generator.py]: Done

File: src/sincere_singularities/modules/order_queue.py [hunk 4]
```diff
@@ -34,7 +35,9 @@
 class OrderQueue:
     """The class for managing the order queue. Orders can be spawned and deleted from here."""
 
-    def __init__(self, interaction: ApplicationCommandInteraction, webhook: Webhook, orders_thread: Thread) -> None:
+    orders_thread: Thread | None = None
```
Comment (koviubi56) [inline src/sincere_singularities/modules/order_queue.py]: Put this in `__init__`

File: src/sincere_singularities/modules/points.py [hunk 5]
```diff
@@ -40,9 +40,7 @@ class TemporaryDatabaseEntry(TypedDict):
 
 
 temporary_database: defaultdict[int, TemporaryDatabaseEntry] = defaultdict(
-    lambda: TemporaryDatabaseEntry(
-        {"points": 0, "restaurants": [load_json("restaurants.json", RestaurantJsonType)[0].name]}
-    )
+    lambda: TemporaryDatabaseEntry({"points": 30, "restaurants": [RESTAURANT_JSON[0].name]})
```
Comment (koviubi56) [inline src/sincere_singularities/modules/points.py]:
```suggestion
    lambda: TemporaryDatabaseEntry({"points": 0, "restaurants": [RESTAURANT_JSON[0].name]})
```

File: src/sincere_singularities/modules/restaurants_view.py [hunk 6]
```diff
@@ -63,24 +65,33 @@ def __init__(self, restaurants: "Restaurants", embeds: list[disnake.Embed], inde
...
         self._prev_page.disabled = self.index == 0
         self._next_page.disabled = self.index == len(self.embeds) - 1
+        if has_restaurant(self.restaurants.interaction.user.id, self.restaurants.all_restaurants[self.index].name):
+            self._enter_restaurant.label = "Enter restaurant"
+        else:
+            self._enter_restaurant.label = "Buy"
+        points = get_points(self.restaurants.interaction.user.id)
+        description = self.embeds[self.index].description
+        assert description
+        self.embeds[self.index].description = re.sub(r"you have \d+", f"you have {points}", description)
```
Comment (koviubi56) [inline src/sincere_singularities/modules/restaurants_view.py:85]: btw doesn't the "You (don't) own this restaurant" text also needs updating?

**Review by koviubi56 — CHANGES_REQUESTED** [review]
If you can fix these, then it's good.

File: pyproject.toml (hunk 1 above)
Comment (Vinyzu) [reply pyproject.toml]: Yes sorry i think i forgot to un-commit this

File: src/sincere_singularities/modules/restaurants_view.py (hunk 6 above)
Comment (Vinyzu) [reply src/sincere_singularities/modules/restaurants_view.py:85]: Fixed by "proxying" the embed property of RestaurantView to Restaurants, which we should have probably done anyways.

**Review by koviubi56 — APPROVED** [review]
LGTM!

**Pull request merged**